### PR TITLE
add llama2 chat template

### PR DESCRIPTION
**Pull request by grimoire**

### CODE REVIEW

**Review by lvhan028 — APPROVED** [review]

**Comment by lvhan028** [comment]
I found the behavior of turbomind tokenizer is different from llama2/tokenizer.
turbomind tokenizer didn't add `bos_token_id` when encoding the first prompt during chating

File: lmdeploy/model.py [hunk 1]
```diff
@@ -125,6 +125,50 @@ def stop_words(self):
...
+
+        Args:
+            prompt (str): user's input prompt
+            sequence_start (bool): indicator for the first round chat of a
+               session sequence
+        Returns:
+            str: the concatenated prompt
+        """
+        if sequence_start:
+            return f'{self.b_inst} ' \
```
Comment (lvhan028) [inline lmdeploy/model.py:160]: May add '<BOS>' in front of `{self.b_inst}`, then the decoded tokens will start with bos_token_id

**Review by lvhan028 — APPROVED** [review]

File: lmdeploy/model.py [hunk 2]
```diff
@@ -125,6 +125,50 @@ def stop_words(self):
...
+class Llama2:
+    """Chat template of LLaMA2 model."""
+
+    def __init__(self):
+
+        B_INST, E_INST = '[INST]', '[/INST]'
+        B_SYS, E_SYS = '<<SYS>>\n', '\n<</SYS>>\n\n'
+
+        DEFAULT_SYSTEM_PROMPT = """
+        You are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe.  Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.
```
Comment (lzhangzz) [inline lmdeploy/model.py:138]: There seems to be a extra newline and extra spaces before the system prompt?

Comment (grimoire) [reply lmdeploy/model.py]: done

**Review by lzhangzz — APPROVED** [review]

**Pull request merged**